mypy/server/deps.py

```diff
@@ -264,6 +283,18 @@ def visit_assignment_stmt(self, o: AssignmentStmt) -> None:
             assert info.typeddict_type is not None
             prefix = '%s.%s' % (self.scope.current_full_target(), info.name())
             self.add_type_dependencies(info.typeddict_type, target=make_trigger(prefix))
+        elif isinstance(rvalue, IndexExpr) and isinstance(rvalue.analyzed, TypeAliasExpr):
+            assert len(o.lvalues) == 1
+            lvalue = o.lvalues[0]
+            assert isinstance(lvalue, NameExpr)
+            attr_trigger = make_trigger('%s.%s' % (self.scope.current_full_target(),
+                                                   lvalue.name))
+            # We need to re-process the target where alias is defined to "refresh" the alias.
+            self.add_type_dependencies(rvalue.analyzed.type)
+            self.add_type_dependencies(rvalue.analyzed.type, attr_trigger)
+            if attr_trigger in self.alias_deps:
+                for dep in self.alias_deps[attr_trigger]:
```

Comment:
I don't understand this. Can you give an example where this is needed? It seems like ast diff should pick up changes to aliases.

Reply:
This is to generate dependencies for the alias name itself, for example this code
```python
A = B[C, D]
```
will generated the following dependencies in `semanal`:
```
python
alias_deps['m'] = {'B', 'C', 'D'}
alias_deps['<A>'] = {'B', 'C', 'D'}
```
The line in question adds the latter.

Again, maybe this is indeed not needed, I will double check now.